Basic simulator body

MiliStefanos opened the pull request:
Simulator has all its parts, simplified

MiliStefanos committed:
Simulator chassis

A first simulator form, that contains the most of its parts simplified.

MiliStefanos committed:
Add class_config

Georacer commented on CMakeLists.txt:
```diff
@@ -1,86 +1,99 @@
+#    set(CMAKE_CXX_FLAGS "-std=c++0x")
+#else()
+#    message(FATAL_ERROR "The compiler ${CMAKE_CXX_COMPILER} has no C++11 support. Please use a different C++ compiler.")
+#endif()
```
Delete them or leave them be, don't comment out code in git.

Georacer commented on CMakeLists.txt:
```diff
@@ -1,86 +1,99 @@
-	sensor_msgs
+  rosgraph_msgs
+  std_msgs
+  std_srvs
```
Please align these elements properly.

Georacer commented on CMakeLists.txt:
```diff
@@ -1,86 +1,99 @@
+  Airdata.msg
+  Control_signals.msg
+  Aero_wrenches.msg
+  Prop_wrenches.msg
```
Use a unified naming scheme for all of these messages. Do not mix camelcase and underscores.

Georacer commented on CMakeLists.txt:
```diff
@@ -1,86 +1,99 @@
+apply_wrench.srv
+model_states_srv.srv
+control_signals_srv.srv
+airdata_srv.srv
```
Fix indentation, each sub-level should be one tab to the right.

Georacer commented on include/Master.hpp:
```diff
@@ -0,0 +1,13 @@
+    // Model model;
+    ros::NodeHandle nh;
+	ros::Subscriber gazebo_sub;
+	Model model;
```
Indent correctly.

Georacer commented on launch/nodes_only.launch:
```diff
@@ -0,0 +1,19 @@
+<launch>
+     
+
+
```
Delete empty lines

Georacer commented on launch/plane.launch:
```diff
@@ -30,6 +30,7 @@
     
     <rosparam command="load" file="$(find last_letter_2)/param/HID.yaml" />
     <rosparam command="load" file="$(find last_letter_2)/param/aerodynamics.yaml" />
+    <rosparam command="load" file="$(find last_letter_2)/param/engine.yaml" />
```
Rename to `propulsion.yaml`

Georacer commented on src/ros/aerodynamics.cpp:
```diff
@@ -0,0 +1,326 @@
+	calc_torques();
+}
+
+void Aerodynamics::calc_additional_data()
```
Rename to calc_air_data.

Georacer replied:
This is a more descriptive name of what this function does.

Georacer commented on src/ros/model.cpp:
```diff
@@ -0,0 +1,193 @@
+	}
+	else
+	{
+		ROS_ERROR("Service down, waiting reconnection...");
```
Please write in a comment that this is a skeleton and it does nothing right now.

Georacer commented on src/ros/model.cpp:
```diff
@@ -0,0 +1,193 @@
+	// call apply_wrench_srv
+	if (sim_step_client.isValid())
+	{
+		if (sim_step_client.call(apply_wrench_srv))
```
Remove they apply_wrench_srv, use an empty message instead.

Georacer commented on src/ros/propulsion.cpp:
```diff
@@ -0,0 +1,73 @@
+	calc_torque();
+}
+
+void Propulsion::calc_additional_data()
```
rename to something more descriptive (e.g. calc_airdata or calc_rel_wind)

Georacer commented on uml/class_model.wsd:
```diff
@@ -0,0 +1,109 @@
+@startuml
```
Delete this file. Noone is expected to keep it up to date. It will quickly become obsolete and misleading.

MiliStefanos committed:
Naming Convention

Fix variables names and convert taps to spaces.

MiliStefanos committed:
Naming Convention 2

MiliStefanos committed:
Environment node + code formatting

Add environment node to code, its parameter file and fix some faults in code formatting

MiliStefanos committed:
Naming Convention 3

Fix msg names

MiliStefanos committed:
Naming Convention 3

Fix msg names

MiliStefanos committed:
Merge remote-tracking branch 'origin/develop' into develop
